pkg/frontend/admin_openshiftcluster_vmresize.go

```diff
@@ -45,44 +43,37 @@ func (f *frontend) _postAdminOpenShiftClusterVMResize(log *logrus.Entry, ctx con
 		return err
 	}
 
-	k, err := f.kubeActionsFactory(log, f.env, doc.OpenShiftCluster)
+	// fetches all the Virtual machines of the Cluster RG
+	vmList, err := action.ListVMsOfResourceGroup(ctx)
 	if err != nil {
 		return err
 	}
+	filteredVmList := f.filterVMList(vmList)
 
-	nodeList, err := k.KubeList(ctx, "node", "")
-	if err != nil {
-		return err
-	}
-
-	var u unstructured.Unstructured
-	var nodes corev1.NodeList
-	if err = json.Unmarshal(nodeList, &u); err != nil {
-		return err
+	// checks if the provided VMname is under the RG
+	if !stringutils.Contains(filteredVmList, vmName) {
+		return api.NewCloudError(http.StatusNotFound, api.CloudErrorCodeNotFound, "",
+			`"The VirtualMachine '%s' under resource group '%s' was not found."`,
+			vmName, resourceGroupName)
 	}
```

Comment:
Thanks for the detailed review, I agree this makes the code a little hard to traverse and read but the reason I chose the current approach instead of taking the suggested one was to leverage having a func ListVMsOfResourceGroup in the azureActions, so it can come in handy at a later date, as we would need a function which would return all the VMs metadata of an RG instead of a name.

Reply:
> so it can come in handy at a later date
I suspected that this was the reason 😄 

Nevertheless, I'm a fervent believer in the YAGNI principle, so keeping the code more complex than necessary for potential additional use cases is something I'm almost always against. IMHO even if you want to keep a function that returns a list for future usage, it is done in a wrong way right now: you should put the function in *azureActions like:
```go
func (a *azureActions) ResourceGroupHasVM(ctx context.Context, vmName string) (bool, error) {
    // etc like in my previous suggestion
```
and you will be reusing  a.virtualMachines.List(ctx, clusterRGName). So the idea is that you don't want the clients of this struct to do the filter by hand (as it happens right now). We should expose the behaviour as a method for them.  I think the suggestion have the best of both worlds as it sticks to the tell don't ask principle but at the same time it gives you the possibility to wrap the internal function (the one that produces the whole list) and expose it to clients in the future by just creating an exported method that internally also calls a.virtualMachines.List(ctx, clusterRGName).

Reply:
Made all the suggested changes. Thanks